compiler/src/dotty/tools/dotc/core/tasty/AttributeUnpickler.scala

```diff
@@ -5,29 +5,16 @@ import scala.language.unsafeNulls
 
 import dotty.tools.tasty.{TastyFormat, TastyReader, TastyBuffer}
 
-import java.nio.charset.StandardCharsets
-
 class AttributeUnpickler(reader: TastyReader):
   import reader._
 
-  lazy val attributeTags: List[Int] =
-    val listBuilder = List.newBuilder[Int]
-    while !isAtEnd do listBuilder += readNat()
-    listBuilder.result()
-
   lazy val attributes: Attributes = {
-    var scala2StandardLibrary = false
-    var explicitNulls = false
-    for attributeTag <- attributeTags do
-      attributeTag match
-        case TastyFormat.SCALA2STANDARDLIBRARYattr => scala2StandardLibrary = true
-        case TastyFormat.EXPLICITNULLSattr => explicitNulls = true
-        case attribute =>
-          assert(false, "Unexpected attribute value: " + attribute)
-    Attributes(
-      scala2StandardLibrary,
-      explicitNulls,
-    )
+    val booleanTags = List.newBuilder[Int]
+
+    while !isAtEnd do
+      booleanTags += readByte()
```

Comment:
Hum, ouch. We shouldn't do that. we should actively refuse to read any unknown tag. Tags may be non-boolean in the future (e.g., source file). This implementation would too easily "fail to fail" on incompatible extensions.

Reply:
I will add non-boolean tags in #18948.